### ISSUE TITLE
[bug][llama_index] Not full tracing captured in llama-index

### ISSUE THREAD

**Issue by diicellman** [Arize-ai/openinference#289]
**Describe the bug**
I have an instance of Arize Phoenix running in a Docker container. I've been using the [instrument.py](https://github.com/Arize-ai/openinference/blob/main/python/examples/llama-index/backend/instrument.py) example for tracing previously, and there were no problems. Today, I pulled the latest Docker container (3.16.0), and now the tracing captures only "chunking" and nothing more. Previously, it captured the full trace for calling query engines, etc.

Here's my specs:
python = 3.10
llama-index = "^0.10.19"
openinference-semantic-conventions = "^0.1.5"
openinference-instrumentation-llama-index = "^1.2.0"
opentelemetry-exporter-otlp = "^1.23.0"
llama-index-readers-telegram = "^0.1.4"
llama-index-llms-anthropic = "^0.1.6"
llama-index-callbacks-arize-phoenix = "^0.1.4"
arize-phoenix = {extras = ["evals"], version = "^3.16.0"}

**To Reproduce**
Steps to reproduce the behavior

**Expected behavior**
To see full tracing
<img width="1598" alt="trace_details_view" src="https://github.com/Arize-ai/openinference/assets/44242534/d66c62ea-0307-431a-97f9-fcb526061268">

**Screenshots**
This is what I get after running query_engine, only "chunking"
<img width="1466" alt="Screenshot 2024-03-18 at 12 26 28" src="https://github.com/Arize-ai/openinference/assets/44242534/eda222a9-f13d-4641-815b-643b88522207">

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version Sonoma 14

**Additional context**

**Comment by mikeldking** [comment]
@diicellman I saw exactly the same regression in llama-index myself. I haven't had time to dive deep into this one but I personally have pinned llma-index to `0.10.19` for the time being. We will investigate further and get back to you but they soft-deprecated callbacks in `0.10.20` - my guess is that this is causing problems.

**Comment by diicellman** [comment]
Thank you for the reply! I was also thinking that llama-index's callbacks updates could be the cause of the issue.

**Comment by mikeldking** [comment]
So far we've reproduced orphaned chunking spans if streams are not consumed. This is caused because the chunking spans are emitted as soon as they are created but the overall trace is not "closed"

![Screenshot 2024-03-21 at 3 24 57 PM](https://github.com/Arize-ai/openinference/assets/5640648/ed70743e-52ad-4602-b3fa-335fedd12ebc)

While not the same as above, it does give us some indication that traces are not shutting down as expected in some situations.

**Comment by mikeldking** [comment]
After some investigation we don’t think the callbacks are working as they previously did. We tried running the LlamaDebugHandler notebook (https://colab.research.google.com/github/jerryjliu/llama_index/blob/main/docs/examples/callbacks/LlamaDebugHandler.ipynb) and we don’t see any traces being printed anymore.
<img width="1117" alt="Screenshot 2024-03-21 at 3 46 06 PM" src="https://github.com/Arize-ai/openinference/assets/5640648/d27cfc46-39d9-47d6-8c9d-95224914f08c">

**Comment by mikeldking** [comment]
@diicellman I just hit this issue again with someone else and just want to double check one thing. It's imperative that `instrument()` is called *BEFORE* any llama-index initialization happens. I'm not sure this will solve anything for you but just wanted to double check since this is after changing when instrumentation is called:

Before: 
<img width="1907" alt="Screenshot 2024-03-21 at 4 49 54 PM" src="https://github.com/Arize-ai/openinference/assets/5640648/4a611efe-7c29-440f-8f57-9399c9f2cf08">

After: 
<img width="1920" alt="Screenshot 2024-03-21 at 5 53 13 PM" src="https://github.com/Arize-ai/openinference/assets/5640648/ad59da28-6136-47b8-909b-c0c7c26438c8">

**Comment by diicellman** [comment]
Thank you for your help! I'm calling the 'instrument()' in the main.py fastapi file on application startup before any llamaindex calls.

**Comment by diicellman** [comment]
I just updated the libraries to latest and I get the same chunking traces and not the full one. I was thinking that maybe the problem is related to async use, because endpoints in my app is async and also the llama-index's `query_engine.aquery()` but then I made them not async and got the same chunking traces.

**Comment by mikeldking** [comment]
> I just updated the libraries to the latest version, and I'm encountering the same chunking traces rather than the full ones. I was considering that perhaps the problem is related to asynchronous usage. Both the endpoints in my app and the llama-index's `query_engine.aquery()` are asynchronous. However, when I made them non-asynchronous, I still got the same chunking traces.

Oh interesting. Is your code on GitHub by chance? Would love to unblock you @diicellman

**Comment by diicellman** [comment]
Yes, my code is on GitHub, but it's in a private repository. If you need to review the code, I can provide the crucial parts that are necessary. I'm sorry for any inconveniences this may cause.

**Comment by mikeldking** [comment]
Moving this to our backlog for now. We've communicated the lack of a trace tree in some contexts to llama-index and they are investigating.

**Comment by mikeldking** [comment]
We will probably fix this via the new instrumentation that is not callbacks
